hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconContainerManager.java

```diff
@@ -423,14 +430,15 @@ public void flushReplicaHistoryMapToDB(boolean clearMap) {
   }
 
   public void upsertContainerHistory(long containerID, UUID uuid, long time,
-                                     long bcsId) {
+                                     long bcsId, String state) {
     Map<UUID, ContainerReplicaHistory> tsMap;
     try {
       tsMap = cdbServiceProvider.getContainerReplicaHistory(containerID);
       ContainerReplicaHistory ts = tsMap.get(uuid);
       if (ts == null) {
         // New entry
-        tsMap.put(uuid, new ContainerReplicaHistory(uuid, time, time, bcsId));
+        tsMap.put(uuid, new ContainerReplicaHistory(uuid, time, time, bcsId,
+            state));
       } else {
         // Entry exists, update last seen time and put it back to DB.
         ts.setLastSeenTime(time);
```

Comment:
Do we need to update the replica state here too?

Reply:
Thanks for the review! Yes, you are right, I'll add that.